### PR TITLE
Add 301s for image reference pages

### PR DESCRIPTION
**Pull request by julienv3**
## Type of change
Sunsetting edu's image reference pages

### What should this PR do?
Add 301 redirects for image reference pages to images.chainguard.dev

### Why are we making this change?
Part of 24g priorities

### What are the acceptance criteria? 
Landing on the images list or individual image reference pages redirects to the appropriate location on images.chainguard.dev

### How should this PR be tested?
Netlify preview deployment doesn't seem to use the nginx config, unlike production?
Tested using curl locally against nginx.

### CODE REVIEW

File: nginx.conf [hunk 1]
```diff
@@ -58,43 +58,27 @@ http {
...
-        "~^/chainguard/chainguard-images/reference/dreamfactory/(.+)?$"  /chainguard/chainguard-images/reference/;
+        # enforce docs turndown
         "~^/chainguard/chainguard-enforce/(.+)?$"  /;
+
+        # reference docs redirects - images
+        "~^/chainguard/chainguard-images/reference/(.+)/image_specs/?$" https://images.chainguard.dev/directory/image/$1/specifications;
+        "~^/chainguard/chainguard-images/reference/(.+)/tags_history/?$" https://images.chainguard.dev/directory/image/$1/versions;
+        "~^/chainguard/chainguard-images/reference/(.+)/provenance_info/?$" https://images.chainguard.dev/directory/image/$1/provenance;
+        "~^/chainguard/chainguard-images/reference/(.+)/$" https://images.chainguard.dev/directory/image/$1/overview;
+        "~^/chainguard/chainguard-images/reference/(.+)$" https://images.chainguard.dev/directory/image/$1/overview;
```
Comment (julienv3) [inline nginx.conf:69]:
```
juverm@jmbp ~ % curl -sIL localhost:8080/chainguard/chainguard-images/reference |egrep '(HTTP|Location)'

HTTP/1.1 301 Moved Permanently
Location: https://images.chainguard.dev/directory
HTTP/2 200 
juverm@jmbp ~ % curl -sIL localhost:8080/chainguard/chainguard-images/reference |egrep '(HTTP|Location)'

HTTP/1.1 301 Moved Permanently
Location: https://images.chainguard.dev/directory
HTTP/2 200 
juverm@jmbp ~ % curl -sIL localhost:8080/chainguard/chainguard-images/reference/ |egrep '(HTTP|Location)'

HTTP/1.1 301 Moved Permanently
Location: https://images.chainguard.dev/directory
HTTP/2 200 
juverm@jmbp ~ % curl -sIL localhost:8080/chainguard/chainguard-images/reference/nginx |egrep '(HTTP|Location)'

HTTP/1.1 301 Moved Permanently
Location: https://images.chainguard.dev/directory/image/nginx/overview
HTTP/2 200 

juverm@jmbp ~ % curl -sIL localhost:8080/chainguard/chainguard-images/reference/nginx/image_specs |egrep '(HTTP|Location)'

HTTP/1.1 301 Moved Permanently
Location: https://images.chainguard.dev/directory/image/nginx/specifications
HTTP/2 200 
juverm@jmbp ~ % curl -sIL localhost:8080/chainguard/chainguard-images/reference/nginx/image_specs/ |egrep '(HTTP|Location)'

HTTP/1.1 301 Moved Permanently
Location: https://images.chainguard.dev/directory/image/nginx/specifications
HTTP/2 200 
juverm@jmbp ~ % curl -sIL localhost:8080/chainguard/chainguard-images/reference/nginx/tags_history/ |egrep '(HTTP|Location)'

HTTP/1.1 301 Moved Permanently
Location: https://images.chainguard.dev/directory/image/nginx/versions
HTTP/2 200 
juverm@jmbp ~ % curl -sIL localhost:8080/chainguard/chainguard-images/reference/nginx/provenance_info/ |egrep '(HTTP|Location)'

HTTP/1.1 301 Moved Permanently
Location: https://images.chainguard.dev/directory/image/nginx/provenance
HTTP/2 200 
juverm@jmbp ~ % curl -sIL localhost:8080/chainguard/chainguard-images/reference/nginx/ |egrep '(HTTP|Location)'           

HTTP/1.1 301 Moved Permanently
Location: https://images.chainguard.dev/directory/image/nginx/overview
HTTP/2 200 
juverm@jmbp ~ % curl -sIL localhost:8080/chainguard/chainguard-images/reference/nginx |egrep '(HTTP|Location)' 

HTTP/1.1 301 Moved Permanently
Location: https://images.chainguard.dev/directory/image/nginx/overview
HTTP/2 200 
juverm@jmbp ~ % 
```

**Comment by erikaheidi** [comment]
Maybe it will be easier / safer to do the redirect first, and remove the content in a separate PR, to avoid these conflicts?

**Comment by erikaheidi** [comment]
Looking good to me! When should we pull the trigger? @julienv3

**Comment by julienv3** [comment]
Checked that 301s were working correctly in staging deploying (thanks @jamonation !)

File: nginx.conf [hunk 2]
```diff
@@ -58,43 +58,36 @@ http {
...
     }
 
     server {
         listen       8080;
         server_name  localhost;
         root   /usr/share/nginx/html/;
 
+        # process $request_uri -> $redirect_url if url is absolute
+        if ($redirect_url ~ ^https://) {
+            rewrite ^(.*)$ $redirect_url permanent;
```
Comment (julienv3) [inline nginx.conf:89]: thanks for the review @erikaheidi !
Just FYI this is ready to merge but I am OOO Jul 15-19, idk how releases work for edu but depending on our level of comfort here we can merge and deploy now or wait til I'm back, up to you?  I expect this to be a binary everything works or everything explodes and having checked in staging with Jamon, it should be the former?

Comment (erikaheidi) [reply nginx.conf:89]: I believe we're good to go ahead, so I will merge. If something explodes, which I don't expect, we can revert it. Thanks @julienv3 !

**Review by erikaheidi — APPROVED** [review]

**Pull request merged**